controllers/etcdpeer_controller.go

```diff
@@ -646,23 +656,35 @@ func (r *EtcdPeerReconciler) Reconcile(ctx context.Context, req ctrl.Request) (_
 		TLS: tlsConfig,
 	}
 
+	// Always requeue after ten seconds, as we don't watch on the membership list. So we don't auto-detect changes made
+	// to the etcd membership API.
+	// TODO(#76) Implement custom watch on etcd membership API, and remove this `requeueAfter`
+	result := ctrl.Result{RequeueAfter: time.Second * 10}
+
+	var healthy bool
+
 	if c, err := r.Etcd.New(etcdConfig); err != nil {
 		log.Error(err, "Unable to connect to etcd")
+		healthy = false
+		r.Recorder.Event(&peer, "Warning", "MemberConnectionFailure", "Cannot connect to etcd member")
+		if r.lastHealthy[peer.Name].IsZero() {
+			log.V(2).Info("Peer has never been healthy")
+		} else if time.Since(r.lastHealthy[peer.Name]) > time.Second*30 {
+			log.Info("Peer unhealthy for more than 30s", "unhealthy_time", time.Since(r.lastHealthy[peer.Name]).Seconds())
+		}
 	} else {
 		defer c.Close()
 		if version, err := c.GetVersion(ctx); err != nil {
 			log.Error(err, "Unable to get Etcd version", "error", err)
 		} else {
 			serverVersion = version.Server
 		}
+		healthy = true
```

Comment:
Very very nitpick but in case the `c.GetVersion` takes a long time for some reason it would be better to have recorded the `lastHealthy` just before that call